editor/src/clj/editor/resource.clj

```diff
@@ -61,18 +71,18 @@
 
 (defn- get-resource-type [workspace resource]
   (let [output-label (if (editable? resource) :resource-types :resource-types-non-editable)
-        resource-types (g/node-value workspace output-label)
+        resource-types (output-label (g/node-by-id workspace))
```

Comment:
We frequently call `resource/resource-type` during compilation, and every call creates an evaluation context. I observed a lot of throw-away atoms (caches in the evaluation context) originating from here. Using basis to lookup the property is much more efficient.

Reply:
Ohh, this should be good!

The program feels more healthy already! ❤️